gustaf/spline/create.py

```diff
@@ -0,0 +1,263 @@
+                "Control Point dimension is smaller than axis dimension,"
+                " filling with zeros"
+            )
+            expansion_dimension = axis.shape[0] - spline.dim
+            cps = np.hstack(
+                (
+                    spline.control_points,
+                    np.zeros((len(spline.control_points), expansion_dimension))
+                )
+            )
+        else:
+            cps = np.copy(spline.control_points)
+
+        # Make sure axis is normalized
+
+        axis_norm = np.linalg.norm(axis)
+        if not np.isclose(axis_norm, 0, atol=settings.TOLERANCE):
+            axis = axis / axis_norm
+        else:
+            raise ValueError("Axis-norm is too close to zero.")
+    else:
+        cps = np.copy(spline.control_points)
+        if spline.control_points.shape[1] == 3:
+            raise ValueError("No rotation axis given")
+
+    # Set Problem dimension
+    problem_dimension = cps.shape[1]
+
+    # Init center
+    if center is not None:
+        center = np.asarray(center).ravel()
+        # Check Axis dimension
+        if not (problem_dimension == center.shape[0]):
+            raise ValueError(
+                "Dimension Mismatch between axis and center of rotation."
+            )
+        cps -= center
+
+    # The parametric dimension is independent of the revolution but the
+    # rotation-matrix is only implemented for 2D and 3D problems
+    if not (cps.shape[1] == 2 or cps.shape[1] == 3):
+        raise NotImplementedError(
+            "Sorry,"
+            "revolutions only implemented for 2D and 3D splines"
+        )
+
+    # Angle must be (0, pi) non including
+    # Rotation is always performed in half steps
+    PI = np.pi
+    round_tolerance = 1e-13
+    minimum_n_knot_spans = int(np.ceil(np.abs((angle + round_tolerance) / PI)))
+    if (n_knot_spans) is None or (n_knot_spans < minimum_n_knot_spans):
+        n_knot_spans = minimum_n_knot_spans
+
+    if "Bezier" in spline.whatami:
+        if n_knot_spans > 1:
+            raise ValueError(
+                "Revolutions are only supported for angles up to 180 degrees"
+                "for Bezier type splines as they consist of only one knot span"
+            )
+
+    # Determine auxiliary values
+    rot_a = angle / (2 * n_knot_spans)
+    half_counter_angle = PI / 2 - rot_a
+    weight = np.sin(half_counter_angle)
+    factor = 1 / weight
+
+    # Assemble rotation matrix
+    if problem_dimension == 2:
+        rotation_matrix = np.array([
+            [np.cos(rot_a), -np.sin(rot_a)],
+            [np.sin(rot_a), np.cos(rot_a)]
+        ]).T
+    else:
+        # See Rodrigues' formula
+        rotation_matrix = np.array(
+            [[0, -axis[2], axis[1]],
+             [axis[2], 0, -axis[0]],
+             [axis[1], axis[0], 0]]
+        )
+        rotation_matrix = (np.eye(3) + np.sin(rot_a) * rotation_matrix + (
+            (1 - np.cos(rot_a)) * np.matmul(rotation_matrix, rotation_matrix))
+        ).T
```

Comment:
```suggestion
        rotation_matrix = (
            np.eye(3)
            + (np.sin(rot_a) * rotation_matrix)
            + ((1 - np.cos(rot_a)) * np.matmul(rotation_matrix, rotation_matrix))
        ).T
```
btw, do you now see the origin of `eye_ders?`